### ISSUE TITLE
Change primary NIC doesn't update the default route

### ISSUE THREAD

**Issue by benjamin-bergia** [joyent/sdc-adminui#191]
I just switched to SDC 20160107. 

Apparently in the new version, when adding a new NIC to a zone, the option to make this new NIC primary is gone. This is kind of annoying but it's fine, since we can still access it through the NIC settings after adding it.

What bother me is that after adding a new NIC and changing the settings for this NIC to be primary, the following happen:
1) The NIC does become primary (I checked with the vmapi)
2) The adminui keep showing the old primary still being primary
3) The default gateway of the zone is still the gateway of the previous primary interface

### Example
1) Initially:
```
sdc-vmapi /vms/80f2973d-a778-4519-9bf0-45ff198e36df | json nics
[
  {
    "interface": "net0",
    "mac": "90:b8:d0:5c:06:9a",
    "vlan_id": 107,
    "nic_tag": "external",
    "gateway": "10.10.7.254",
    "gateways": [
      "10.10.7.254"
    ],
    "netmask": "255.255.255.0",
    "ip": "10.10.7.68",
    "ips": [
      "10.10.7.68/24"
    ],
    "network_uuid": "2afe7b88-166a-4e6d-8b8e-24b0b56ca3b7",
    "mtu": 1500,
    "primary": true
  }
]
```
```
netstat -rn

Routing Table: IPv4
  Destination           Gateway           Flags  Ref     Use     Interface 
-------------------- -------------------- ----- ----- ---------- --------- 
default              10.10.7.254          UG        1          0 net0      
10.0.0.0             10.10.7.254          UG        1          0 net0      
10.10.7.0            10.10.7.68           U         3         39 net0      
127.0.0.1            127.0.0.1            UH        2          0 lo0       

Routing Table: IPv6
  Destination/Mask            Gateway                   Flags Ref   Use    If   
--------------------------- --------------------------- ----- --- ------- ----- 
::1                         ::1                         UH      2       0 lo0   
```

2) After adding and setting the new NIC as primary

```
netstat -rn

Routing Table: IPv4
  Destination           Gateway           Flags  Ref     Use     Interface 
-------------------- -------------------- ----- ----- ---------- --------- 
default              10.10.7.254          UG        1          0 net0      
10.0.0.0             10.10.7.254          UG        1          0 net0      
10.10.7.0            10.10.7.68           U         3         38 net0      
x.x.x.x                 x.x.x.y      U         2          0 net1      
127.0.0.1            127.0.0.1            UH        2          0 lo0       

Routing Table: IPv6
  Destination/Mask            Gateway                   Flags Ref   Use    If   
--------------------------- --------------------------- ----- --- ------- ----- 
::1                         ::1                         UH      2       0 lo0   
```

```
sdc-vmapi /vms/80f2973d-a778-4519-9bf0-45ff198e36df | json nics
[
  {
    "interface": "net0",
    "mac": "90:b8:d0:5c:06:9a",
    "vlan_id": 107,
    "nic_tag": "external",
    "gateway": "10.10.7.254",
    "gateways": [
      "10.10.7.254"
    ],
    "netmask": "255.255.255.0",
    "ip": "10.10.7.68",
    "ips": [
      "10.10.7.68/24"
    ],
    "network_uuid": "2afe7b88-166a-4e6d-8b8e-24b0b56ca3b7",
    "mtu": 1500
  },
  {
    "interface": "net1",
    "mac": "90:b8:d0:d7:b2:11",
    "vlan_id": 20,
    "nic_tag": "external",
    "gateway": "x.x.x.y",
    "gateways": [
      "x.x.x.y"
    ],
    "netmask": "255.255.255.240",
    "ip": "x.x.x.x",
    "ips": [
      "x.x.x.x/28"
    ],
    "network_uuid": "7c9339b2-d613-4a53-b815-096b1d16b279",
    "mtu": 1500,
    "primary": true
  }
]
```

**Comment by benjamin-bergia** [comment]
Quick update. I did the same operation on an other freshly provisioned Zone and the UI is fine. The new interface is correctly displayed as primary. Unfortunately the default route is still wrong.

**Comment by askfongjojo** [comment]
Hi @benjamin-bergia, have you rebooted the zone after the primary nic change? If not, please try that and see if the default route is updated after zone restart. Based on the outcome, I'll follow up with an appropriate fix (vmapi bug or usability improvement). Thanks!

**Comment by benjamin-bergia** [comment]
@askfongjojo Just tested on a different zone. Restarting the zone after switching the interface to primary does fix the route issue AND the UI issue.

Would it be possible to also get back the "Make this the primary NIC" option back when adding a new NIC ?

**Comment by askfongjojo** [comment]
Hi @benjamin-bergia - I just confirmed that the option got dropped inadvertently during a bug fix. We're going to put it back. You can track the status of the ticket for it here: https://smartos.org/bugview/ADMINUI-2279.

Sorry for the inconvenience caused.

**Comment by benjamin-bergia** [comment]
Cool !
Thank you

**Comment by askfongjojo** [comment]
This has been fixed in release-20160218 (commit ea52b92).
